mariner/file_formats/cbddlp.py

```diff
@@ -0,0 +1,192 @@
+    skip_3: int = StructType.uint32()
+    skip_4: int = StructType.uint32()
+    skip_5: int = StructType.uint32()
+    skip_6: int = StructType.uint32()
+    machine_offset: int = StructType.uint32()
+    machine_size: int = StructType.uint32()
+    encryption_mode: int = StructType.uint32()
+    time_seconds: int = StructType.uint32()
+    unknown_01: int = StructType.uint32()
+    version_patch: int = StructType.unsigned_char()
+    version_minor: int = StructType.unsigned_char()
+    version_major: int = StructType.unsigned_char()
+    version_release: int = StructType.unsigned_char()
+    unknown_02: int = StructType.uint32()
+    unknown_03: int = StructType.uint32()
+    unknown_04: float = StructType.float32()
+    unknown_05: int = StructType.uint32()
+    unknown_06: int = StructType.uint32()
+    unknown_07: float = StructType.float32()
+
+
+@dataclass(frozen=True)
+class CBDDLPLayerDef(LittleEndianStruct):
+    layer_height_mm: float = StructType.float32()
+    layer_exposure: float = StructType.float32()
+    layer_off_time: float = StructType.float32()
+    image_offset: int = StructType.uint32()
+    image_length: int = StructType.uint32()
+    unknown_01: int = StructType.uint32()
+    image_info_size: int = StructType.uint32()
+    unknown_02: int = StructType.uint32()
+    unknown_03: int = StructType.uint32()
+
+
+@dataclass(frozen=True)
+class CBDDLPPreview(LittleEndianStruct):
+    resolution_x: int = StructType.uint32()
+    resolution_y: int = StructType.uint32()
+    image_offset: int = StructType.uint32()
+    image_length: int = StructType.uint32()
+
+
+REPEAT_RGB15_MASK: int = 1 << 5
+
+
+def _read_image(width: int, height: int, data: bytes) -> png.Image:
+    array: List[List[int]] = [[]]
+
+    (i, x) = (0, 0)
+    while i < len(data):
+        color16 = int(struct.unpack_from("<H", data, i)[0])
+        i += 2
+        repeat = 1
+        if color16 & REPEAT_RGB15_MASK:
+            repeat += int(struct.unpack_from("<H", data, i)[0]) & 0xFFF
+            i += 2
+
+        (r, g, b) = (
+            (color16 >> 0) & 0x1F,
+            (color16 >> 6) & 0x1F,
+            (color16 >> 11) & 0x1F,
+        )
+
+        while repeat > 0:
+            array[-1] += [r, g, b]
+            repeat -= 1
+
+            x += 1
+            if x == width:
+                x = 0
+                array.append([])
+
+    array.pop()
+
+    return png.from_array(array, "RGB;5")
+
+
+@dataclass(frozen=True)
+class CBDDLPFile:
+    filename: str
+    bed_size_mm: Tuple[float, float, float]
+    height_mm: float
+    layer_height_mm: float
+    layer_count: int
+    resolution: Tuple[int, int]
+    print_time_secs: int
+    end_byte_offset_by_layer: Sequence[int]
+    slicer_version: str
+    printer_name: str
+
+    @classmethod
+    def read(self, path: pathlib.Path) -> "CBDDLPFile":
+        with open(str(path), "rb") as file:
+            CBDDLP_header = CBDDLPHeader.unpack(file.read(CBDDLPHeader.get_size()))
```

Comment:
nit picking here, but can you rename this as `cbddlp_header` so this follows [PEP8](https://www.python.org/dev/peps/pep-0008/#function-and-variable-names) where all variables should have only lowercase characters?